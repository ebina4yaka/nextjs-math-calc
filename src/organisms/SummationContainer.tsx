import React from 'react'
import SummationFormula from '../molecules/SummationFormula'
import SummationInputField from '../molecules/SummationInputField'
import useSummation from '../hooks/useSummation'
import BackButton from '../atoms/BackButton'

export default function SummationContainer(): React.ReactElement {
  const { n, k, answer, setNewSummation } = useSummation()
  return (
    <>
      <SummationFormula n={n} k={k} answer={answer} />
      <SummationInputField n={n} k={k} setNewSummation={setNewSummation} />
      <BackButton />
    </>
  )
}
