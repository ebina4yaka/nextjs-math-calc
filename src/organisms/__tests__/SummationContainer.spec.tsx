import React from 'react'
import renderer from 'react-test-renderer'
import SummationContainer from '../SummationContainer'

test('Summation Snapshot Test', () => {
  const tree = renderer.create(<SummationContainer />).toJSON()
  expect(tree).toMatchSnapshot()
})
