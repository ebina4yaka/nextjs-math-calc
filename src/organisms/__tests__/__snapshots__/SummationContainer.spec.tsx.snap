// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summation Snapshot Test 1`] = `
Array [
  <div
    className="makeStyles-container-1"
  >
    <div>
      <div
        className="makeStyles-number-3"
      >
        10
      </div>
      <div
        className="makeStyles-sigma-2"
      >
        Î£
      </div>
      <div
        className="makeStyles-number-3"
      >
        k = 
        1
      </div>
    </div>
    <div
      className="makeStyles-number-3"
    >
      k
    </div>
    <div
      className="makeStyles-answer-4"
    >
      <div
        className="makeStyles-number-3"
      >
         = 
        55
      </div>
    </div>
  </div>,
  <div
    className="makeStyles-container-5"
  >
    <div
      className="MuiFormControl-root MuiTextField-root makeStyles-textField-6"
    >
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
        onClick={[Function]}
      >
        <div
          className="MuiInputAdornment-root MuiInputAdornment-positionStart"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
          >
            n=
          </p>
        </div>
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
          disabled={false}
          id="n"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="input-n"
          required={false}
          type="number"
          value={10}
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root makeStyles-textField-6"
    >
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
        onClick={[Function]}
      >
        <div
          className="MuiInputAdornment-root MuiInputAdornment-positionStart"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
          >
            k=
          </p>
        </div>
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
          disabled={false}
          id="k"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="input-k"
          required={false}
          type="number"
          value={1}
        />
      </div>
    </div>
  </div>,
  <div
    className="makeStyles-root-7"
  >
    <a
      aria-disabled={false}
      aria-label="back"
      className="MuiButtonBase-root MuiIconButton-root"
      href="/"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </svg>
        Back
      </span>
    </a>
  </div>,
]
`;
